lib/PuppeteerSharp.Tests/Target/Tests.cs

```diff
@@ -0,0 +1,183 @@
+﻿using Microsoft.AspNetCore.Http;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace PuppeteerSharp.Tests.Target
+{
+    [Collection("PuppeteerLoaderFixture collection")]
+    public class Tests : PuppeteerPageBaseTest
+    {
+        [Fact]
+        public void BrowserTargetsShouldReturnAllOfTheTargets()
+        {
+            // The pages will be the testing page and the original newtab page
+            var targets = Browser.Targets();
+            Assert.Contains(targets, target => target.Type == "page"
+                && target.Url == TestConstants.AboutBlank);
+            Assert.Contains(targets, target => target.Type == "other"
+                && target.Url == string.Empty);
+        }
+
+        [Fact]
+        public async Task BrowserPagesShouldReturnAllOfThePages()
+        {
+            // The pages will be the testing page and the original newtab page
+            var allPages = (await Browser.Pages()).ToArray();
+            Assert.Equal(2, allPages.Length);
+            Assert.Contains(Page, allPages);
+            Assert.NotSame(allPages[0], allPages[1]);
+        }
+
+        [Fact]
+        public async Task ShouldBeAbleToUseTheDefaultPageInTheBrowser()
+        {
+            // The pages will be the testing page and the original newtab page
+            var allPages = await Browser.Pages();
+            var originalPage = allPages.First(p => p != Page);
+            Assert.Equal("Hello world", await originalPage.EvaluateExpressionAsync<string>("['Hello', 'world'].join(' ')"));
+            Assert.NotNull(await originalPage.GetElementAsync("body"));
+        }
+
+        [Fact]
+        public async Task ShouldReportWhenANewPageIsCreatedAndClosed()
+        {
+            var otherPageTaskCompletion = new TaskCompletionSource<PuppeteerSharp.Page>();
+            async void TargetCreatedEventHandler(object sender, TargetChangedArgs e)
+            {
+                otherPageTaskCompletion.SetResult(await e.Target.Page());
+                Browser.TargetCreated -= TargetCreatedEventHandler;
+            }
+            Browser.TargetCreated += TargetCreatedEventHandler;
+            await Page.EvaluateFunctionHandle("url => window.open(url)", TestConstants.CrossProcessUrl);
```

Comment:
I know this is not part of this PR, but this should be `EvaluateFunctionHandleAsync`